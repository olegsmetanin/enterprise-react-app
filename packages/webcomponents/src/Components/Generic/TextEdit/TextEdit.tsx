import * as React from 'react'

export interface TextEditProps {
  value: any
  onChange?: (e) => void
};

export class TextEdit extends React.Component<TextEditProps, void>{
  render() {
    return (
      <input
        type="text"
        value={this.props.value}
        onChange={(e) => this.props.onChange(e)}
      />
    );
  }
}