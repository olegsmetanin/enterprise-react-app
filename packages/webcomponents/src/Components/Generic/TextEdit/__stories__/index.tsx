/* tslint:disable:no-unused-variable */
import * as React from 'react'
/* tslint:disable:no-unused-variable */
import {storiesOf} from '@kadira/storybook';
import {TextEdit}  from '../TextEdit'

declare var module: any;

interface IWrap {
  value: any
}

class Wrap extends React.Component<void, IWrap> {
  constructor(props) {
    super(props)
    this.state = {
      value: 'qwe'
    }
  }

  render() {
    return (
      <TextEdit
        value={this.state.value}
        onChange={(e) => this.setState({value: e.target.value})}
      />
    )
  }

}

storiesOf('TextEdit', module)
  .add('Default', () => (
    <Wrap/>
  ));