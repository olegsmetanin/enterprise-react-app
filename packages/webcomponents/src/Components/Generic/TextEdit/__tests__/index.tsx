/* tslint:disable:no-unused-variable */
import * as React from 'react'
/* tslint:disable:no-unused-variable */
import * as renderer from 'react-test-renderer'

import {TextEdit}  from '../TextEdit'

it('renders correctly', () => {
    const tree = renderer.create(
        <TextEdit value="qwe"/>
    ).toJSON()
    expect(tree).toMatchSnapshot()
})