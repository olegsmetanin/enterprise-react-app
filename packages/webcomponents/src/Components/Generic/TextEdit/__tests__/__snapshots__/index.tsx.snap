exports[`test renders correctly 1`] = `
<input
  onChange={[Function]}
  type="text"
  value="qwe" />
`;
